lib/items/tool-bar-button-view.js

```diff
@@ -79,6 +62,10 @@ class ToolBarButtonView extends ToolBarItem {
 
   /** Add an icon for the button using built-in icons. */
   addIcon () {
+    if (!this.options.icon) {
+      return;
+    }
+
```

Comment:
I am not personally a fan of this type of refactoring. I deliberately wrote the code like this. I prefer checking the options outside of the function, and then only call the function if necessary.

Reply:
Let me say up front that I'm okay with reverting this changes if you don't agree with the following :slightly_smiling_face: 

Firstly, it is not at all uncommon to check if the input is valid in other (including typed) programming languages, many programming languages have something similar to a `null` value that you nearly always need to take into account when writing a function.

Second, regarding the input/output type stability. This is not at all a problem in this case, as the function does not return anything. And even if it should return something, this pattern is still preferred over checking validity at the function call if it is possible to return a default value that always makes sense (sometimes called a [null object](https://en.wikipedia.org/wiki/Null_object_pattern)).

Lastly, the overhead of calling a few functions a couple of times (the number of buttons in the toolbar to be exact) is not something we as programmers working in a high-level language like JavaScript should worry about in comparison to making code readable and maintainable. first because the overhead is very low and second because the compiler or runtime environment can optimize it.

Reply:
I agree with your points that this may not be applicable to this situation particularly and we don't see the issues I stated, but this is a general style that I always follow. If you want to write the checks compact you do something like this too:
```js
this.options.icon ?  this.addIcon() : ;
```

Reply:
I'm even less of a  fan of writing it compactly in that way, as it obfuscates what you're trying to achieve (if you want to use an if-statement, use an if-statement and not a ternary operator).

If the original way is the way you always write it and that's how you want to do it here, that's completely fine by me. Let me know here and I will update the PR.

However, if you agree with my points I'm not sure why you would stick with our original style. But that's a discussion that should not block this PR from getting merged 🙂

Reply:
In this case, this is just a minor styling change. I will let this to be up to you.

Reply:
In that case let's merge it like this :smiley: